sysdig/resource_sysdig_secure_posture_policy.go

```diff
@@ -0,0 +1,430 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	err = d.Set(SchemaIDKey, policy.ID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaNameKey, policy.Name)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaDescriptionKey, policy.Description)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaVersionKey, policy.Description)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaTypeKey, policy.Type)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaLinkKey, policy.Link)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaMinKubeVersionKey, policy.MinKubeVersion)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaMaxKubeVersionKey, policy.MaxKubeVersion)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaIsActiveKey, policy.IsActive)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set(SchemaPlatformKey, policy.Platform)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Set groups
+	if err := setGroups(d, policy.RequirementsGroup); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func setGroups(d *schema.ResourceData, groups []v2.RequirementsGroup) error {
+	var groupsData []interface{}
+	for _, group := range groups {
+		groupData := map[string]interface{}{
+			"id":          group.Id,
+			"name":        group.Name,
+			"description": group.Description,
+		}
+
+		// Recursively set nested groups and requirements
+		if len(group.Requirements) > 0 {
+			requirementsData := setRequirements(group.Requirements)
+			groupData["requirement"] = requirementsData
+		}
+		if len(group.Folders) > 0 {
+			nestedGroupsData := setGroups(d, group.Folders)
+			groupData["group"] = nestedGroupsData
+		}
+
+		groupsData = append(groupsData, groupData)
+	}
+	fmt.Println("groupsData: ", groupsData)
+	return d.Set(SchemaGroupKey, groupsData)
+}
+
+func setRequirements(requirements []v2.Requirement) []interface{} {
+	var requirementsData []interface{}
+	for _, req := range requirements {
+		reqData := map[string]interface{}{
+			"id":          req.Id,
+			"name":        req.Name,
+			"description": req.Description,
+		}
+
+		// Set controls for each requirement
+		if len(req.Controls) > 0 {
+			controlsData := setControls(req.Controls)
+			reqData["control"] = controlsData
+		}
+
+		requirementsData = append(requirementsData, reqData)
+	}
+	return requirementsData
+}
+
+func setControls(controls []v2.Control) []interface{} {
+	var controlsData []interface{}
+	for _, ctrl := range controls {
+		ctrlData := map[string]interface{}{
+			"name":    ctrl.Name,
+			"enabled": ctrl.Enabled,
+		}
+		controlsData = append(controlsData, ctrlData)
+	}
+	return controlsData
+}
+func resourceSysdigSecurePosturePolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// TODO: implement deletion
```

Comment:
Is it fine to have non implemented delete?

Reply:
At the moment, we support only create and update operations. We are currently working on the implementation of delete functionality